### PR TITLE
MISUV-6855: enable disabled test in WhatYouOweResponseAuditModelSpec & other related changes 

### PR DESCRIPTION
**Pull request by avelx**
* set valid fixedDate day[3];
  * remove LocalDate.now usages
[Remaining usages of LocalDate.now to be investigate ](https://jira.tools.tax.service.gov.uk/browse/MISUV-6871)
* upgrade sbt-auto-build to recent~ version;

### CODE REVIEW

File: test/services/NextUpdatesServiceSpec.scala [hunk 1]
```diff
@@ -399,27 +402,5 @@ class NextUpdatesServiceSpec extends TestSupport with MockObligationsConnector w
       )
     }
 
-    "Fix for MISUV-6494" in new Setup {
```
Comment (avelx) [inline test/services/NextUpdatesServiceSpec.scala:402]: >removed as we assume this is not applicable anymore

File: test/audit/models/WhatYouOweResponseAuditModelSpec.scala [hunk 2]
```diff
@@ -77,74 +74,73 @@ class WhatYouOweResponseAuditModelSpec extends TestSupport {
       testWhatYouOweResponseAuditModel().auditType shouldBe auditEvent
     }
 
-// TODO: re-enable unit test
```
Comment (avelx) [inline test/audit/models/WhatYouOweResponseAuditModelSpec.scala:80]: test :wrench:

File: test/testConstants/BusinessDetailsTestConstants.scala [hunk 3]
```diff
@@ -39,8 +39,10 @@ object BusinessDetailsTestConstants {
   val address = AddressModel("8 Test", Some("New Court"), Some("New Town"), Some("New City"), Some("NE12 6CI"), "United Kingdom")
   val businessIncomeSourceId = "1234"
 
+  val fixedDate : LocalDate = LocalDate.of(2023, 12, 15)
+
   val getCurrentTaxYearEnd: LocalDate = {
-    val currentDate: LocalDate = LocalDate.now
+    val currentDate: LocalDate = fixedDate
```
Comment (simondrugan16) [inline test/testConstants/BusinessDetailsTestConstants.scala:45]: :wrench: perhaps rename this as it's not longer the current date

**Comment by avelx** [comment]
:smiley: Please review this PR test

File: test/testUtils/TestSupport.scala [hunk 4]
```diff
@@ -78,7 +79,21 @@ trait TestSupport extends UnitSpec with GuiceOneAppPerSuite with BeforeAndAfterE
   implicit val conf: Configuration = app.configuration
   implicit val environment: Environment = app.injector.instanceOf[Environment]
   implicit val appConfig: FrontendAppConfig = app.injector.instanceOf[FrontendAppConfig]
-  implicit val dateService: DateService = app.injector.instanceOf[DateService]
+
+  // Set fixed date for DateService
+  lazy val fixedDate : LocalDate = LocalDate.of(2023, 12, 15)
+  implicit val dateService: DateService = new DateService {
+
+    override def getCurrentDate(isTimeMachineEnabled: Boolean): LocalDate = fixedDate
```
Comment (simondrugan16) [inline test/testUtils/TestSupport.scala:87]: ❓ same as before, will we update this every year/taxyear? If not it's not really *current*Date. Same for methods below

Comment (avelx) [reply test/testUtils/TestSupport.scala:87]: this is in scope of test implementation of the dateService; we are fixing date to be static for the test, so its fine

File: test/testConstants/BusinessDetailsTestConstants.scala (hunk 3 above)
Comment (avelx) [reply test/testConstants/BusinessDetailsTestConstants.scala:45]: fixed

Comment (simondrugan16) [reply test/testConstants/BusinessDetailsTestConstants.scala]: uhh it's still called *currentDate* for me

File: test/testUtils/TestSupport.scala (hunk 4 above)
Comment (avelx) [reply test/testUtils/TestSupport.scala:87]: :memo: => getCurrentDate is defined in the main, this is not something we can change;
we just providing value which is required in the scope of tests etc;
Have a look at the implementation in the main ->
/income-tax-view-change-frontend/app/services/DateService.scala

File: test/testConstants/BusinessDetailsTestConstants.scala (hunk 3 above)
Comment (avelx) [reply test/testConstants/BusinessDetailsTestConstants.scala]: looks like some gitHub glitch, check one more time

File: test/testUtils/TestSupport.scala (hunk 4 above)
Comment (simondrugan16) [reply test/testUtils/TestSupport.scala:87]: gotcha

**Pull request merged**